src/openrct2/Game.cpp

```diff
@@ -129,17 +129,24 @@ GAME_COMMAND_CALLBACK_POINTER* game_command_callback_get_callback(uint32_t index
 
 void game_increase_game_speed()
 {
-    gGameSpeed = std::min(gConfigGeneral.debugging_tools ? 5 : 4, gGameSpeed + 1);
-    if (gGameSpeed == 5)
-        gGameSpeed = 8;
+    gGameSpeed++;
+    if (gGameSpeed == GAMESPEED_HYPER && !gConfigGeneral.debugging_tools)
+        gGameSpeed--;
+
+    if (game_is_paused())
+        pause_toggle();
```

Comment:
I'm not a fan of this. This means that the game speed cannot be changed while the game remains paused, since it will unpause right away. I'd leave the pause state to the player.